app/components/DappRequest/InvokeFunction/InvokeFunction.jsx

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable no-nested-ternary */
+// @flow
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import classNames from 'classnames'
+import { wallet as n3Wallet } from '@cityofzion/neon-js-next'
+
+import {
+  useWalletConnectWallet,
+  TSession,
+  TSessionRequest,
+} from '@cityofzion/wallet-connect-sdk-wallet-react'
+import { NeonInvoker } from '@cityofzion/neon-invoker'
+import ConnectionLoader from '../../ConnectDapp/ConnectionLoader'
+import { ROUTES, WITNESS_SCOPE } from '../../../core/constants'
+import CloseButton from '../../CloseButton'
+import FullHeightPanel from '../../Panel/FullHeightPanel'
+import WallletConnect from '../../../assets/icons/wallet_connect.svg'
+import styles from '../../../containers/ConnectDapp/styles.scss'
+import DialogueBox from '../../DialogueBox'
+import WarningIcon from '../../../assets/icons/warning.svg'
+import Confirm from '../../../assets/icons/confirm_connection.svg'
+import Deny from '../../../assets/icons/deny_connection.svg'
+import Tooltip from '../../Tooltip'
+import Info from '../../../assets/icons/info.svg'
+
+import { convertToArbitraryDecimals } from '../../../core/formatters'
+import { getNode, getRPCEndpoint } from '../../../actions/nodeStorageActions'
+import Invocation from './Invocation'
+import InvokeResult from './InvokeResult'
+
+const electron = require('electron')
+
+type Props = {
+  request: TSessionRequest,
+  session: TSession,
+  isHardwareLogin: boolean,
+  publicKey: string,
+  wif: string,
+  history: any,
+  showSuccessNotification: ({ message: string }) => any,
+  theme: string,
+  net: string,
+}
+
+const InvokeFunction = ({
+  request,
+  session,
+  isHardwareLogin,
+  history,
+  publicKey,
+  wif,
+  showSuccessNotification,
+  theme,
+  net,
+}: Props) => {
+  const requestParams = useMemo(() => request.params.request.params, [request])
+
+  const { rejectRequest, approveRequest } = useWalletConnectWallet()
+  const [loading, setLoading] = useState(false)
+  const [fee, setFee] = useState()
+  const [transactionHash, setTransactionHash] = useState()
+  const [errorMessage, setErrorMessage] = useState()
+
+  const handleCalculateFee = useCallback(
+    async () => {
+      try {
+        setLoading(true)
+        const account = new n3Wallet.Account(isHardwareLogin ? publicKey : wif)
```

Comment:
I could be wrong but I dont think we ever need to instantiate the account with `wif` when performing `invokeFunction` 👀